velox/functions/lib/MapFromEntries.cpp

```diff
@@ -219,12 +224,20 @@ class MapFromEntriesFunction : public exec::VectorFunction {
           nulls, indices, decodedRowVector->size(), rowVector->childAt(1));
     }
 
-    // To avoid creating new buffers, we try to reuse the input's buffers
-    // as many as possible.
+    // For Presto, need construct map vector based on input nulls for possible
+    // outer expression like try(). For Spark, use the updated nulls unless it's
+    // empty.
+    if (throwOnNull_) {
+      nulls = inputArray->nulls();
+    } else {
+      if (decodedRowVector->size() == 0) {
+        nulls = inputArray->nulls();
```

Comment:
Could you please provide more details on the difference of `inputArray->nulls()` and `the updated nulls` when `decodedRowVector->size() == 0`?